pkg/controller/controller.go

```diff
@@ -449,7 +459,7 @@ func (p *csiProvisioner) Provision(options controller.ProvisionOptions) (*v1.Per
 	if options.PVC.Spec.DataSource != nil && (rc.clone || rc.snapshot) {
 		volumeContentSource, err := p.getVolumeContentSource(options)
 		if err != nil {
-			return nil, fmt.Errorf("error getting handle for DataSource Type %s by Name %s: %v", options.PVC.Spec.DataSource.Kind, options.PVC.Spec.DataSource.Name, err)
+			return nil, controller.ProvisioningNoChange, fmt.Errorf("error getting handle for DataSource Type %s by Name %s: %v", options.PVC.Spec.DataSource.Kind, options.PVC.Spec.DataSource.Name, err)
```

Comment:
This is possibly more a clarification than a request for change, why are the following series of errors `ProvisioningNoChange` rather than `ProvisioningFinished`?

We have not issued the `CreateVolume` and there is an error, do we want the provisioner to retry, because these errors are recoverable?

Reply:
Because this may be second iteration of provisioning. The first one failed in CSI driver with `ProvisioningInBackground`. The provisioner tries for second time and now  due to a network hiccup it cannot e.g. get a Secret. But the provisioning is still in progress from the last attempt, hence `NoChange` and the provisioner must try again until it succeeds.

Which may be tricky when someone e.g. has deleted the Secret. So there it is still possible to leak a volume, but only when admin actively screws something up.

Reply:
Ack! thank you for the explanation.